02_knight_and_princess/anton_potapov/graph_create.cpp

```diff
@@ -0,0 +1,167 @@
+#include <cassert>
+#include <fstream>
+#include <iostream>
+#include <map>
+#include <set>
+#include <sstream>
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+using VertexId = int;
+using EdgeId = int;
+
+class Vertex {
+ public:
+  explicit Vertex(const VertexId& vertex_id) : id_(vertex_id) {}
+  std::string get_json_string() const {
+    std::stringstream json_stringstream;
+    json_stringstream << "{\"id\":" << id_ << ","
+                      << "\"edge_ids\":[";
+    for (auto it = adjacted_edges_.begin(); it != adjacted_edges_.end(); ++it) {
+      json_stringstream << *it;
+      if (std::next(it) != adjacted_edges_.end()) {
+        json_stringstream << ",";
+      }
+    }
+    json_stringstream << "]}";
+    return json_stringstream.str();
+  }
+  bool is_edge_adjacted(const EdgeId& edge_id) const {
+    return adjacted_edges_.find(edge_id) != adjacted_edges_.end();
+  }
+  void add_edge(const EdgeId& edge_id) {
+    assert(!is_edge_adjacted(edge_id) &&
+           "edge that is to be added already exists");
+    adjacted_edges_.insert(edge_id);
+  }
+
+ private:
+  const VertexId id_;
+  std::set<EdgeId> adjacted_edges_;
+};
+
+class Edge {
+ public:
+  Edge(const EdgeId& edge_id, const VertexId& vertex1, const VertexId& vertex2)
+      : id_(edge_id), vertex1_id_(vertex1), vertex2_id_(vertex2) {}
+  VertexId get_first_vertex() const { return vertex1_id_; }
+  VertexId get_second_vertex() const { return vertex2_id_; }
+  std::string get_json_string() const {
+    std::stringstream json_stringstream;
+    json_stringstream << "{\"id\":" << id_ << ","
+                      << "\"vertex_ids\":[" << vertex1_id_ << "," << vertex2_id_
+                      << "]}";
+    return json_stringstream.str();
+  }
+
+ private:
+  const EdgeId id_;
+  const VertexId vertex1_id_, vertex2_id_;
+};
+
+class Graph {
+ public:
+  bool is_vertex_exists(const VertexId& vertex) const {
+    return vertices_.find(vertex) != vertices_.end();
+  }
+
+  bool is_connected(const VertexId& vertex1, const VertexId& vertex2) const {
+    for (const auto& edge : edges_) {
+      if ((edge.second.get_first_vertex() == vertex1 &&
+           edge.second.get_second_vertex() == vertex2) ||
+          (edge.second.get_first_vertex() == vertex2 &&
+           edge.second.get_second_vertex() == vertex1)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  VertexId add_vertex() {
+    const VertexId new_vertex_id = next_vertex_id_;
+    update_next_vertex_id_();
```

Comment:
nit: Эту логику можно было ещё улучшить и энкапсулировать всю логику связанную с `id` внутри функции `update_next_vertex_id_()`
```suggestion
    const VertexId new_vertex_id = get_next_vertex_id();
```
Но это мелочи